src/main/app-updater.ts

```diff
@@ -1,20 +1,71 @@
-import { autoUpdater } from "electron-updater";
+import { autoUpdater, UpdateInfo } from "electron-updater";
 import logger from "./logger";
+import { isDevelopment, isTestEnv } from "../common/vars";
+import { delay } from "../common/utils";
+import { areArgsUpdateAvailableToBackchannel, AutoUpdateLogPrefix, broadcastMessage, onceCorrect, UpdateAvailableChannel, UpdateAvailableToBackchannel } from "../common/ipc";
+import * as uuid from "uuid";
+import { ipcMain } from "electron";
 
-export class AppUpdater {
-  static readonly defaultUpdateIntervalMs = 1000 * 60 * 60 * 24; // once a day
+function handleAutoUpdateBackChannel(event: Electron.IpcMainEvent, ...[arg]: UpdateAvailableToBackchannel) {
+  if (arg.doUpdate) {
+    if (arg.now) {
+      logger.info(`${AutoUpdateLogPrefix}: User chose to update now`);
+      autoUpdater.downloadUpdate()
+        .then(() => autoUpdater.quitAndInstall())
+        .catch(error => logger.error(`${AutoUpdateLogPrefix}: Failed to download or install update`, { error }));
+    } else {
+      logger.info(`${AutoUpdateLogPrefix}: User chose to update on quit`);
+      autoUpdater.autoInstallOnAppQuit = true;
+      autoUpdater.downloadUpdate()
+        .catch(error => logger.error(`${AutoUpdateLogPrefix}: Failed to download update`, { error }));
+    }
+  } else {
+    logger.info(`${AutoUpdateLogPrefix}: User chose not to update`);
+  }
+}
 
-  static checkForUpdates() {
-    return autoUpdater.checkForUpdatesAndNotify();
+/**
+ * starts the automatic update checking
+ * @param interval milliseconds between interval to check on, defaults to 24h
+ */
+export function startUpdateChecking(interval = 1000 * 60 * 60 * 24): void {
+  if (isDevelopment || isTestEnv) {
+    return;
   }
 
-  constructor(protected updateInterval = AppUpdater.defaultUpdateIntervalMs) {
-    autoUpdater.logger = logger;
+  autoUpdater.logger = logger;
+  autoUpdater.autoDownload = false;
+  autoUpdater.autoInstallOnAppQuit = false;
+
+  autoUpdater
+    .on("update-available", (args: UpdateInfo) => {
+      try {
+        // use a UUID so that this back-channel is harder to discover
+        const backchannel = uuid.v4();
```

Comment:
Why do we need "hidden" backchannel? Could backchannel be just  something like `auto-update:${version}`.

Reply:
This is to try and discourage malicious extensions from spamming this backchannel

Reply:
How could extensions spam this? They don't have access to this IPC.

Reply:
Do extensions not have access to `ipcRenderer`?

Reply:
Lens is not exposing any `ipc` stuff to extensions. When (if) we expose `ipc` to extensions it most likely will be prefixed so that extensions cannot cause trouble to Lens internal communication (like stores).

Reply:
Okay I can change over to an incrementing channel